tests/unit/tracking.tcl

```diff
@@ -780,6 +780,81 @@ start_server {tags {"tracking network"}} {
         r debug pause-cron 0
     } {OK} {needs:debug}
 
+    foreach resp {3 2} {
+        test "RESP$resp based basic invalidation with client reply off" {
+            clean_all
+
+            $rd hello $resp
+            $rd read
+            $rd client tracking on
+            $rd read
+
+            $rd_sg set foo bar
+            $rd get foo
+            $rd read
+
+            $rd client reply off
+
+            $rd_sg set foo bar2
+
+            if {$resp == 3} {
+                assert_equal {invalidate foo} [$rd read]
+            }
```

Comment:
this is actually a NOP, just a coverage (i thought it is nice to have in case the future changes), i will improve the comment later.
```
} else {
        /* If are here, the client is not using RESP3, nor is
         * redirecting to another client. We can't send anything to
         * it since RESP2 does not support push messages in the same
         * connection. */
        if (!(old_flags & CLIENT_PUSHING)) c->flags &= ~CLIENT_PUSHING;
        return;
    }
```

Reply:
when we expect some reply to be skipped, it'a usually a good idea to add a PING (expecting PONG) afterwards.

Reply:
yes. you mean add a PING in these tracking tests?
`client reply off` will disable all reply, so i did not add the ping in the first place, it require a `client reply on`. now we add another `client reply` tests, so i think this is ok

Reply:
yes, i think that in all these tracking tests where we did REPLY OFF, we must verify things didn't get messed up and no unexpected reply was pushed to the client and we missed it because we didn't finish with a `read`.
we can't do a `read` since it'll hang (unless there's a bug), so instead we need to finish with a REPLY ON and then a PING.

regarding this specific test, that entire test doesn't make much sense in RESP2, right, (not just the line at the bottom).
we can keep it, but let's change the comment to mention it better.
i.e. something like "this entire test is mostly irrelevant for RESP2, but we run it anyway just for some extra coverage".

Reply:
thanks, got it, updated it